app/src/main/java/space/taran/arknavigator/mvp/model/repo/preview/generator/VideoPreviewGenerator.kt

```diff
@@ -0,0 +1,55 @@
+package space.taran.arknavigator.mvp.model.repo.preview.generator
+
+import android.graphics.Bitmap
+import android.net.Uri
+import android.util.Log
+import java.nio.file.Path
+import space.taran.arknavigator.ui.App
+import wseemann.media.FFmpegMediaMetadataRetriever
+
+object VideoPreviewGenerator : PreviewGenerator() {
+    override val acceptedExtensions: Set<String> =
+        setOf("mp4", "avi", "mkv", "mov", "wmv", "flv", "webm", "ts", "mpg")
+    override val acceptedMimeTypes: Set<String>
+        get() = setOf()
+
+    override fun generate(path: Path, previewPath: Path, thumbnailPath: Path) {
+        val preview = generatePreview(path)
+        preview?.let {
+            storePreview(previewPath, it)
```

Comment:
Why not just `storePreview(previewPath, preview)`?

Reply:
`preview` is `nullable` as `FFmpegMediaMetadataRetriever` is not sure about preview image and return null if image is not there.